### PR TITLE
Big upgrade 

### PR DESCRIPTION
**Pull request by QIvan**
catching up on the work started here https://github.com/Netflix/hollow-reference-implementation/pull/16 I upgraded everything up to the latest and greatest version
fixed: 
* out-of-date hollow version
* gradle build now is able to build with java 11+
* use gradle plugin for consumer code generation 
* use maven plugin for consumer code generation (atm works, but generates out-of-date code due old version in the plugin's dependencies. see https://github.com/IgorPerikov/hollow-maven-plugin/pull/17) 
* other dependencies were out of date too 
* minor code fixes

special thanks to @liufuyang and @avietrov

### CODE REVIEW

File: src/main/java/how/hollow/producer/SourceDataRetriever.java [hunk 1]
```diff
@@ -15,7 +15,10 @@
  *     limitations under the License.
  *
  */
-package how.hollow.producer.datamodel;
+package how.hollow.producer;
```
Comment (Sunjeet) [inline src/main/java/how/hollow/producer/SourceDataRetriever.java:18]: nit: clean up unused imports

**Review by Sunjeet — COMMENTED** [review]
Thanks for getting this repo up to date. One nitpicky comment, rest LGTM!

File: src/main/java/how/hollow/producer/SourceDataRetriever.java (hunk 1 above)
Comment (QIvan) [reply src/main/java/how/hollow/producer/SourceDataRetriever.java:18]: done

**Comment by QIvan** [comment]
thank you @Sunjeet for the review! 
I optimized the imports, I guess it's ready to merge now :+1:

**Pull request merged**